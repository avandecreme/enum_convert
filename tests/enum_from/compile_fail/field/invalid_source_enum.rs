use enum_convert::EnumFrom;

enum Source {
    Struct { x: i32 },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from]
    Struct {
        // Should be #[enum_from(Source::Struct.x)]
        #[enum_from(NonExistent::Struct.x)]
        a: i32,
    },
}

fn main() {}
