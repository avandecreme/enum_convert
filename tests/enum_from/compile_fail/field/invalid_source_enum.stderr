error: Field mapping for unknown enum
  --> tests/enum_from/compile_fail/field/invalid_source_enum.rs:13:21
   |
13 |         #[enum_from(NonExistent::Struct.x)]
   |                     ^^^^^^^^^^^
